### PR TITLE
Add loopless FBA

### PR DESCRIPTION
**Pull request by stelmo**
Add a basic variant of thermodynamic FBA. Not the mixed interger approach, but rather the quadratically constrained implementation.

### CODE REVIEW

**Comment by cylon-x** [comment]
:green_circle: &nbsp;Commit 8a770138 is formatted properly.

**Comment by stelmo** [comment]
This is basically done, the only problem is that the function is nonlinear :/

**Comment by cylon-x** [comment]
:green_circle: &nbsp;Commit 51c7e8b6 is formatted properly.

**Comment by cylon-x** [comment]
:green_circle: &nbsp;Commit b06be994 is formatted properly.

**Comment by cylon-x** [comment]
:green_circle: &nbsp;Commit 25646753 is formatted properly.

**Comment by cylon-x** [comment]
:green_circle: &nbsp;Commit 606a203a is formatted properly.

**Comment by cylon-x** [comment]
:red_square: &nbsp;Commit 02979d87 requires formatting!

Required formatting changes summary:
```
 src/base/utils/Reaction.jl | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)
```

**Comment by cylon-x** [comment]
:red_square: &nbsp;Commit 02979d87 requires formatting!

Required formatting changes summary:
```
 src/base/utils/Reaction.jl | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)
```

**Comment by cylon-x** [comment]
:red_square: &nbsp;Commit 1e9461a1 requires formatting!

Required formatting changes summary:
```
 test/analysis/fba_with_crowding.jl | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)
```

**Comment by cylon-x** [comment]
:green_circle: &nbsp;Commit 2195a017 is formatted properly.

**Comment by cylon-x** [comment]
:green_circle: &nbsp;Commit 06a40266 is formatted properly.

**Comment by cylon-x** [comment]
:green_circle: &nbsp;Commit 56424af7 is formatted properly.

**Comment by cylon-x** [comment]
:green_circle: &nbsp;Commit c2f6a4de is formatted properly.

**Comment by cylon-x** [comment]
:green_circle: &nbsp;Commit c4845e4b is formatted properly.

File: src/base/constants.jl [hunk 1]
```diff
@@ -9,7 +9,8 @@ const _constants = (
     tolerance = 1e-6,
     sampling_keep_iters = 100,
     sampling_size = 1000,
-    exchange_suffixes = ["_e", "[e]", "(e)"],
+    strict_inequality_tolerance = 1,
```
Comment (exaexa) [inline src/base/constants.jl:12]: naming: perhaps add something about what purpose this serves, like "loopless_inequality_tolerance" or "slack_variable_bound_tolerance" or something?

File: test/analysis/fba_with_crowding.jl [hunk 2]
```diff
@@ -1,13 +1,16 @@
 @testset "FBA with crowding constraints" begin
     model = load_model(model_paths["e_coli_core.json"])
-    idxs = find_internal_reactions(model)
```
Comment (exaexa) [inline test/analysis/fba_with_crowding.jl:3]: semantic shift very welcome

**Review by exaexa — CHANGES_REQUESTED** [review]
Tiny changes required, otherwise looks OK, well balanced add/remove line ratio :D

**Comment by cylon-x** [comment]
:green_circle: &nbsp;Commit b79b8516 is formatted properly.

**Comment by cylon-x** [comment]
:green_circle: &nbsp;Commit 5edca7f7 is formatted properly.

**Comment by cylon-x** [comment]
:green_circle: &nbsp;Commit 29769c75 is formatted properly.

**Comment by stelmo** [comment]
I squashed into two commits: one for the NLP stuff that didn't work (to be continued :]) and the actual implemented one: loopless.

**Review by exaexa — APPROVED** [review]

**Pull request merged**